### ISSUE TITLE
Feature Request: Support small form factor devices

### ISSUE THREAD

**Issue by hendrix2222** [mcmonkeyprojects/SwarmUI#39]
### Feature Idea

The UI is currently designed for use on desktop devices, and is rendered appropriately for that use-case.

For non-desktop devices such as tablets or other mobile devices, the UI is not rendered in a user-friendly way. Some support for at least basic text-to-image generation on small form factors would be greatly appreciated.

**Comment by mcmonkey4eva** [comment]
Yep, mobile support on the Generate tab is definitely a major feature requirement.

I have a rough idea of how it'd work:

The main screen should be just the prompt box and image result area,
and a button on the left to pull up the parameters list, which pops up to cover the screen and you can edit then swipe it back away
similarly you can pull up the other menus (image history / model browser / etc) and they also temporarily take over the full screen while you use them.

Basically take all the current UI features that are placed into cells, and swap them to being sub-screens to swap between, with the 'main prompt + result' area as the centerpoint

**Comment by hendrix2222** [comment]
Yeah, I'm a backend dev by trade, so UI work isn't something I'm even remotely qualified to speak to, other than it's a lot more difficult than it looks to get things to display nicely across different form factors. And I think you sort of alluded to it, but it may actually end up making the desktop render better by making things a bit more modular anyway.

Separately, I had made a feature request here: https://github.com/ShiftHackZ/Stable-Diffusion-Android/issues/204

to potentially add support for SwarmUI's API to their native Android application, but I have no clue how active the development is there, etc.

**Comment by jonathan-f-silva** [comment]
While this is not properly addressed, I did some (not so) quick hack in the simple tab to at least do some gens on mobile. Here are the changes if anyone is interested:
https://github.com/mcmonkeyprojects/SwarmUI/compare/mcmonkeyprojects:SwarmUI:master...jonathan-f-silva:SwarmUI:feat/simple-mobile?expand=1

<details>
<summary>Demo GIF</summary>

![simple-mobile](https://github.com/mcmonkeyprojects/SwarmUI/assets/85591570/066f913e-2515-49b8-81bf-3f34039189c4)

</details>

**Comment by mcmonkey4eva** [comment]
If you get it clean and it doesn't break on desktop sized usage, I'd love a PR for that!

**Comment by mcmonkey4eva** [comment]
mobile support is in the works, Swarm now has extremely minimal but functional phone screen support. Opens to the center area only, swipe from the edges to pull in the side views.
